app/controllers/communicarts_controller.rb

```diff
@@ -24,7 +24,10 @@ def send_cart
   end
 
   def approval_reply_received
-    cart = Cart.where(external_id: params['cartNumber'].to_i).where(status: 'pending').first
+    cart = Cart.joins(:proposal).where(
+      external_id: params['cartNumber'].to_i,
```

Comment:
Not every Proposal will have one, but I could see it being useful for enough use cases that we should keep it in there. On the other hand, it might be more clear if it's on the use-case-specific tables, e.g. the `cl_number` (?) for NCR, or the `concur_voucher_id` if we were handling travel requests. Thoughts?